### PR TITLE
Helm chart with ingress

### PR DESCRIPTION
**Pull request by mariusandra**
Hi!

This PR includes an updated and cleaned helm chart and many other fixes. Here are the changes I made and some questions/comments regarding them.

1) I modified `settings.py` and added a new way to configure your database. The combined `DATABASE_URL` format doesn't work well with kubernetes, as the database password is stored in a secret store and given directly to the app. Now the app will also accept `POSTHOG_DB_NAME`, `POSTHOG_DB_USER`, `POSTHOG_DB_PASSWORD`, `POSTHOG_POSTGRES_HOST` and `POSTHOG_POSTGRES_PORT`. In case you merge this PR and publish a new docker image, the helm (and gitlab) configuration will be much simpler. Now I need to manually *also* give a `DATABASE_URL` env variable just to make it work.

2) Is there a reason why `collectstatic` is not part of the `production.Dockerfile`, but executed in `./bin/docker`? I've seen it as part of dockerfiles in other django projects... and it seems weird to run this same step every time a web pod is started. The `preview.Dockerfile` file even skips this step it seems...?

3) I'm now running the `gunicorn` command manually when starting the web pod and skipping the `./bin/docker` script. That's because the migrations are now run in a separate helm/k8s worker that is started every time the chart is installed or upgraded. When starting 2+ web pods and letting them both run "migrate", weird things could happen as well, so it makes sense to keep it separate. It might make sense to make a new `./bin/docker-runserver` script that both the helm chart and the `./bin/docker` files will call. I would have done it, but I'm just waiting for an answer regarding if the collectstatic script is needed there as well or not...

4) Right now the chart just pulls down the docker image `posthog/posthog:latest`. However [this is not a good practice](https://kubernetes.io/docs/concepts/containers/images/#updating-images) and it's recommended to actually update the chart every time a new version of your app is out. I'm not sure what's the best way to do that though. Manually editing some files and committing the changes every time you tag a new release I suppose?

The fields that need to be adapted are 
- in `chart/Chart.yaml` -> `appVersion = "1.0.10"` and `version += 0.0.1`
- in `chart/values.yaml` -> `image.pullPolicy = "IfNotPresent"` and `image.tag = "1.0.10"`

5) When deploying the chart locally, I seem to run into an issue when sometimes loading `/_health/` or any other URL times out the first time... and then after a while starts working. This means that the readiness probe for the web pods also fails and they get restarted a few times until it finally works.

Any idea what could be causing this? The database is ready and the migrations have ran. Everything seems to be in order, yet somehow the app just keeps loading until the request times out.

I hope it's just something to do with my local configuration, yet it always just happens in the first minutes after the app starts (or the chart is upgraded) and then seems to be magically fixed afterwards. 

6) There's now also a README.md file with the chart that describes all the values and their defaults.

### CODE REVIEW

**Comment by timgl** [comment]
Great work as per!

1. Happy with that.
2. I don't think there's a strong reason for this so happy for this to be moved to the dockerfile itself
3. Good catch on the migrations. Again think that's the right way to go.
4. We do automatically build a `posthog/posthog:release-[version]` every time we push a new version. [There's a checklist](https://github.com/PostHog/posthog/wiki/Release-new-version) for how to release a new version of PostHog, we could make this a step in that.
5. Hm Django does need to load a few things before it properly gets started and that does take a couple of seconds. Is that consistent with what you've seen or is it longer than that?

File: chart/README.md [hunk 1]
```diff
@@ -1,7 +1,157 @@
-Run the following in the root of the posthog folder to install the chart and run posthog with an internal database
+# PostHog
 
-```shell script
-  helm install -f chart/values-local.yaml posthog ./chart
+[PostHog](https://posthog.io/) is a cross-platform crash reporting and aggregation platform.
```
Comment (timgl) [inline chart/README.md]: It says posthog.io throughout. I'm not even sure we own that domain ;-)

File: gunicorn.config.py [hunk 2]
```diff
@@ -22,6 +22,6 @@ def on_starting(server):
 """
 \x1b[0m
 """)
-    print("Server running on \x1b[4mhttp://127.0.0.1:8000\x1b[0m")
+    print("Server running on \x1b[4mhttp://0.0.0.0:8000\x1b[0m")
```
Comment (timgl) [inline gunicorn.config.py:25]: While we're at it, `'{}:{}'.format(*server.address[0])` will actually give the correct address all the time

**Comment by timgl** [comment]
As discussed, due to different licensing of the source of these Helm charts we're moving this to [posthog/charts](https://github.com/posthog/charts)

**Comment by mariusandra** [comment]
I moved the chart and made a new PR (#431) for the other changes. Thus closing.

**Pull request closed without merging**